src/test/java/soolutions/hometaskapp/DepositAccountTest.java

```diff
@@ -0,0 +1,35 @@
+package soolutions.hometaskapp;
+
+import org.junit.jupiter.api.Test;
+
+import soolutions.hometaskapp.account.Account;
+import soolutions.hometaskapp.account.BasicAccount;
+import soolutions.hometaskapp.core.Amount;
+import soolutions.hometaskapp.user.BasicUser;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DepositAccountTest {
+    @Test
+    void depositAccount() {
+        final Account account = new BasicAccount(new BasicUser());
+        account.deposit(new Amount(0, "USD"));
+        assertEquals(new Amount(0, "USD"), account.balance());
+        account.deposit(new Amount(1, "USD"));
+        assertEquals(new Amount(1, "USD"), account.balance());
+        account.deposit(new Amount(10.99, "USD"));
+        assertEquals(new Amount(11.99, "USD"), account.balance());
+    }
+
+    @Test
+    void depositAccount_currency_mismatch() {
```

Comment:
Did you mean to name this test `depositAmount_currency_mismatch()`? 

Also this kind of seems like a strange mix of snake case and camel case, maybe we should stick to one or the other?